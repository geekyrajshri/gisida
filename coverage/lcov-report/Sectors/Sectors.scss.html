<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Sectors/Sectors.scss</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">Sectors</a> Sectors.scss
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">.sectors-menu {
  position: fixed;
  width: 300px;
  height: 100%;
  color: #555;
  z-index: 1000;
  overflow-y: scroll;
  background: #F7F6EF;
  box-shadow: 0 1px 7px rgba(0, 0, 0, 0.4);
&nbsp;
  .close-btn {
    position: absolute;
    right: 15px;
    top: 15px;
    z-index: 1000;
    color: #555;
    display: block;
  }
    
  ul {
    list-style-type: none;
  }
  
  .sectors {
    padding: 0;
    overflow-y: scroll;
    
    li.sector a {
      color: #555;
      display: block;
      font-weight: 600;
      padding: 25px 15px;
      background: rgba(255,255,255, 0.4);
      border-bottom: 1px solid #eee;
      
      &amp;:hover,
      &amp;:focus {
        text-decoration: none;
        color: #555;
      }
    }
    
  }
  
 .layers {
   padding: 20px 0 20px 20px;
   display: none;
   
    li.layer {
      position: relative;
      display: block;
      width: 100%;
    }
 }
  
  label {
    color: #555;
    font-weight: normal;
    font-size: 12px;
    width: 80%;
    display: block;
    margin: 0 0 10px 20px;
  }
  
  input[type='checkbox'] {
    position: absolute;
    left: 0;
    top: -2px;
  }
  
}
&nbsp;
.open-btn {
  position: fixed;
  display: block;
  color: #333;
  background: #fff;
  border: 2px solid #ddd;
  border-radius: 5px;
  width: 32px;
  line-height:2;
  font-size: 15px;
  text-align: center;
  top: 10px;
  display: none;
}
&nbsp;
#sector-menu-1-wrapper .open-btn {
  left: 90px;
}
#sector-menu-2-wrapper .open-btn {
  left: 53%;
}
&nbsp;
&nbsp;
#sector-menu-1 {
  height: 100%;
  top: 0;
  left: 80px;
}
#sector-menu-2 {
  height: 100%;
  top: 0;
  left: 52%;
  display: none;
}</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jun 27 2017 15:27:56 GMT+0300 (EAT)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
