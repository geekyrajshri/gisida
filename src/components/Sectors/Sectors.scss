.sectors-menu {
  position: fixed;
  width: 300px;
  height: 100%;
  color: #555;
  z-index: 1000;
  overflow-y: scroll;
  background: #f7f6ef;
  box-shadow: 0 1px 7px rgba(0, 0, 0, 0.4);
  left: 80px;

  .sectors {
    padding: 0;
    overflow-y: scroll;

    li.sector {
      border-bottom: 1px solid #eee;

      a {
        color: #555;
        display: block;
        font-weight: 600;
        padding: 10px 15px;

        &:hover,
        &:focus {
          text-decoration: none;
          color: #555;
        }
      }
    }
  }
}

.framework-sectors-menu {
  padding-top: 5px;
  position: fixed;
  width: 18%;
  height: 95%;
  color: #555;
  z-index: 1000;
  overflow-y: auto;
  background: #fff;
  box-shadow: 0 1px 7px rgba(0, 0, 0, 0.4);
  top: 6%;
  left: 0;

  .sectors {
    padding: 0;
    overflow-y: auto;

    li.sector {
      border-bottom: 1px solid #eee;
      a {
        color: #cb181d;
        display: block;
        font-weight: 600;
        padding: 10px 15px;

        &:hover,
        &:focus {
          text-decoration: none;
          color: #cb181d;
        }
      }
    }
  }
}

.close-btn {
  position: absolute;
  right: 15px;
  top: 15px;
  z-index: 1000;
  color: #555;
  display: block;
}

ul {
  list-style-type: none;
}

.filters-menu {
  height: 100%;
  top: 10em;
  left: 0;
}

.layers {
  padding: 20px 0 20px 18px;
  display: none;

  li {
    position: relative;
    display: block;
    width: 100%;

    label {
      color: #555;
      font-weight: normal;
      font-size: 12px;
      width: 80%;
      display: block;
      margin: 0 0 10px 20px;
      text-indent: -23px;
    }

    input[type='checkbox'],
    input[type='radio'] {
      margin-right: 5%;
    }
  }
}

.open-btn {
  position: fixed;
  color: #333;
  background: #fff;
  border: 2px solid #ddd;
  border-radius: 5px;
  width: 32px;
  line-height: 2;
  font-size: 15px;
  text-align: center;
  top: 10px;
  display: none;
}

.framework-open-btn {
  position: fixed;
  color: #333;
  background: #fff;
  border: 2px solid #ddd;
  border-radius: 5px;
  width: 32px;
  line-height: 2;
  font-size: 15px;
  text-align: center;
  top: 3em;
  left: 1em;
  display: none;
}

#sector-menu-1-wrapper .open-btn {
  left: 90px;
}

#sector-menu-2-wrapper .open-btn {
  left: 53%;
}

.sectors-menu-wrapper {
  position: fixed;
}

#sector-menu-2 {
  height: 100%;
  top: 0;
  left: 52%;
  display: none;
}
