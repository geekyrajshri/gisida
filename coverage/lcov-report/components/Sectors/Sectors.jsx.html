<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Sectors/Sectors.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/Sectors</a> Sectors.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
require('./Sectors.scss');
import Layers from '../Layers/Layers';
&nbsp;
const Sectors = ({ sectorMenuId, mapTargetId, sectorData, layerData, onToggleSectors = f =&gt; f, onSectorClick = f =&gt; f, onLayerChange = f =&gt; f }) =&gt;
  (&lt;div id={`${sectorMenuId}-wrapper`} className="sectors-menu-wrapper"&gt;
    &lt;a href="#" onClick={e =&gt; onToggleSectors(e)} className="open-btn"&gt;&lt;span className="glyphicon glyphicon-list" /&gt;&lt;/a&gt;
    &lt;div id={sectorMenuId} className="sectors-menu"&gt;
      &lt;a className="close-btn" onClick={e =&gt; onToggleSectors(e)} href="#"&gt;&lt;span className="glyphicon glyphicon-remove" /&gt;&lt;/a&gt;
      &lt;ul className="sectors"&gt;
        { sectorData.map((sector, i) =&gt;
    (&lt;li className="sector" key={i}&gt;&lt;a href="#" onClick={e =&gt; onSectorClick(e)}&gt;{ sector.sector } &lt;span className="caret" /&gt;&lt;/a&gt;
      &lt;Layers onLayerChange={onLayerChange} mapTargetId={mapTargetId} layers={sector.layers} layerData={layerData} /&gt;
    &lt;/li&gt;))
  }
      &lt;/ul&gt;
    &lt;/div&gt;
  &lt;/div&gt;);
&nbsp;
export default Sectors;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jun 27 2017 15:17:21 GMT+0300 (EAT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
