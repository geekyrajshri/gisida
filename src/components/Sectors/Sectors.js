require('./Sectors.scss');
import Layers from '../Layers/Layers'

const Sectors = ({ sectorMenuId, mapTargetId, sectorData, layerData, onToggleSectors=f=>f, onSectorClick=f=>f, onLayerChange=f=>f }) =>
 <div id={sectorMenuId + '-wrapper'} className="sectors-menu-wrapper">
 <a href="#" onClick={(e) => onToggleSectors(e)} className="open-btn"><span className="glyphicon glyphicon-list"></span></a>
 <div id={sectorMenuId} className="sectors-menu">
  <a className="close-btn" onClick={(e) => onToggleSectors(e)} href="#"><span className="glyphicon glyphicon-remove"></span></a>
  <ul className="sectors">
  { sectorData.map((sector, i) => 
    <li className="sector" key={i}><a href="#" onClick={(e) => onSectorClick(e)}>{ sector.sector } <span className="caret"></span></a>
     <Layers onLayerChange={onLayerChange} mapTargetId={ mapTargetId } layers={ sector.layers } layerData={ layerData } />
    </li>)
  } 
  </ul>
 </div>
</div>

export default Sectors