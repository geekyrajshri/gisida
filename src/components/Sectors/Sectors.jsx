import React from 'react';
import PropTypes from 'prop-types';
import Layers from '../Layers/Layers';
import Filters from '../Filters/Filters';

require('./Sectors.scss');

const Sectors = ({ sectorMenuId,
  mapTargetId,
  sectorData,
  layerData,
  onToggleSectors = f => f,
  onToggleView = f => f,
  onSectorClick = f => f,
  onLayerChange = f => f,
  onFilterSelect = f => f,
  view,
  defaultView,
  filters,
  selected,
}) =>
  (<div id={`${sectorMenuId}-wrapper`} className="sectors-menu-wrapper">
    <a
      href="#"
      onClick={e => onToggleSectors(e)}
      className={
        view === 'framework' || defaultView === 'framework' ? 'framework-open-btn' : 'open-btn'}
    ><span className="glyphicon glyphicon-list" /></a>
    <div
      className={view === 'framework' || defaultView === 'framework' ?
        'framework-sectors-menu' : 'sectors-menu'}
    >
      <a className="close-btn" onClick={e => onToggleSectors(e)} href="#"><span className="glyphicon glyphicon-remove" /></a>
      <ul className="sectors">
        {sectorData.map((sector, i) =>
          // eslint-disable-next-line react/no-array-index-key
          ((sector.filters || sector.views) ?
            (<li className="sector" key={i}><a href="#" onClick={e => onSectorClick(e)}>{sector.sector}<span className="caret" /></a>
              <Filters
                sector={sector.sector}
                filters={sector.filters}
                headers={sector.headers}
                views={sector.views}
                onToggleView={onToggleView}
                onFilterSelect={onFilterSelect}
                checked={filters}
                currentView={view}
              />
            </li>) :
            view !== 'framework' ?
              (<li className="sector" key={i}><a
                href="#"
                onClick={e => onSectorClick(e)}
              >{sector.sector}<span className="caret" /></a>
                <Layers
                  onLayerChange={onLayerChange}
                  mapTargetId={mapTargetId}
                  layers={sector.layers}
                  defaultView={defaultView}
                  headers={sector.headers}
                  layerData={layerData}
                  onFilterSelect={onFilterSelect}
                  selected={selected}
                  sector={sector}
                />
              </li>) : ''))
        }
      </ul>
    </div>
  </div>);


Sectors.propTypes = {
  sectorMenuId: PropTypes.string.isRequired,
  mapTargetId: PropTypes.string.isRequired,
  sectorData: PropTypes.arrayOf(PropTypes.any).isRequired,
  layerData: PropTypes.objectOf(PropTypes.any).isRequired,
  onToggleSectors: PropTypes.func.isRequired,
  onToggleView: PropTypes.func.isRequired,
  onSectorClick: PropTypes.func.isRequired,
  onLayerChange: PropTypes.func.isRequired,
  view: PropTypes.string.isRequired,
  defaultView: PropTypes.string.isRequired,
  onFilterSelect: PropTypes.func,
  filters: PropTypes.arrayOf(PropTypes.any),
  selected: PropTypes.arrayOf(PropTypes.any),
};

Sectors.defaultProps = {
  onFilterSelect: null,
  filters: null,
  selected: null,
};

export default Sectors;
