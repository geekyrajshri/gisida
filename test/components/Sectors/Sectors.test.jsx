import React from 'react';
import renderer from 'react-test-renderer';
import Sectors from '../../../src/components/Sectors/Sectors';
import layerData from '../../fixtures/layers.json';

describe('Sectors', () => {
    const toggleSectors = jest.fn();
    const sectorClick = jest.fn();
    const changeLayer = jest.fn();
    const toggleView = jest.fn();
    const sectorData = [{ sector: 'Sample Sector', layers: ['test_layer'], filters: ['test_filter'], views: ['test_views']}];
    it('Sectors renders sectors.json clusters', () => {
        const component = renderer.create(
            <Sectors
                sectorMenuId="sector-menu-1"
                mapTargetId='map-1'
                onToggleSectors={toggleSectors}
                onSectorClick={sectorClick}
                onLayerChange={changeLayer}
                onToggleView={toggleView}
                sectorData={sectorData}
                layerData={layerData}
                view='currentView'
                defaultView='testView'
            />);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});
