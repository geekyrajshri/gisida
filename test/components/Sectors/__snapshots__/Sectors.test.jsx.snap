// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sectors Sectors renders sectors.json clusters 1`] = `
<div
  className="sectors-menu-wrapper"
  id="sector-menu-1-wrapper"
>
  <a
    className="open-btn"
    href="#"
    onClick={[Function]}
  >
    <span
      className="glyphicon glyphicon-list"
    />
  </a>
  <div
    className="sectors-menu"
  >
    <a
      className="close-btn"
      href="#"
      onClick={[Function]}
    >
      <span
        className="glyphicon glyphicon-remove"
      />
    </a>
    <ul
      className="sectors"
    >
      <li
        className="sector"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Sample Sector
          <span
            className="caret"
          />
        </a>
        <ul
          className="filters"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <li
            className="filter"
          >
            <input
              checked={false}
              id="test_views"
              name="switch-view"
              onClick={[Function]}
              type="radio"
              value="test_views"
            />
            <label
              htmlFor="test_views"
            >
              test_views
            </label>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;
